app/js/controllers/rules_controller.js

```diff
@@ -36,6 +42,25 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
   $scope.$watch('ordering_str', function(value) {
     $scope.ordering = value.value.split(',');
   });
+
+  $scope.$watch('pr_ch_filter', function(value) {
+    $scope.pr_ch_selected = value.split(',');
+  });
+
+  $scope.pr_ch_options = ["All rules"];
+  Rules.getProducts().success(function(response_prs) {
+    Rules.getChannels().success(function(response_chs) {
+      response_prs.product.forEach(function(pr) {
+        response_chs.channel.forEach(function(ch) {
+          if (!ch.contains("-") && !ch.contains("*") && $scope.pairExists(pr, ch)){
```

Comment:
I think what might be happening is that this new block of code is always executed in this controller, but $scope.pairExists only exists if rule_id isn't set (it's in the else block of the if/else that starts at line 8). Maybe this should be in that else block as well? This filter probably doesn't matter for a single rule view.